include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -1035,6 +531,567 @@ class post_layout_optimization_impl
+                               {
+
+                                   // add fanout to the respective route
+                                   add_fanout_to_route(op, fanouts_set.empty(), ffd);
+                                   add_fanout_to_route(fanout, fanouts_set.empty(), ffd);
+
+                                   // continue until gate or primary output (PO) is found
+                                   while (lyt.is_wire_tile(fanout) && lyt.fanout_size(lyt.get_node(fanout)) != 0 &&
+                                          lyt.fanout_size(lyt.get_node(fanout)) != 2)
+                                   {
+                                       ffd.to_clear.push_back(fanout);
+                                       fanout = lyt.outgoing_data_flow(fanout).front();
+
+                                       // add fanout to the respective route
+                                       add_fanout_to_route(fanout, fanouts_set.empty(), ffd);
+                                   }
+
+                                   // set the respective fanout based on the route
+                                   if (fanouts_set.empty())
+                                   {
+                                       fanout1 = fanout;
+                                   }
+                                   else
+                                   {
+                                       fanout2 = fanout;
+                                   }
+
+                                   fanouts_set.insert(fanout);
+                               }
+                           });
+
+        // add fanins and fanouts if existing
+        if (!fanin1.is_dead())
+        {
+            ffd.fanins.push_back(fanin1);
+        }
+        if (!fanin2.is_dead())
+        {
+            ffd.fanins.push_back(fanin2);
+        }
+        if (!fanout1.is_dead())
+        {
+            ffd.fanouts.push_back(fanout1);
+        }
+        if (!fanout2.is_dead())
+        {
+            ffd.fanouts.push_back(fanout2);
+        }
+
+        return ffd;
+    }
+    /**
+     * This helper function computes a path between two coordinates using the A* algorithm.
+     * It then obstructs the tiles along the path in the given layout.
+     *
+     * @param lyt Reference to the layout.
+     * @param start_tile The starting coordinate of the path.
+     * @param end_tile The ending coordinate of the path.
+     * @return The computed path as a sequence of coordinates in the layout.
+     */
+    layout_coordinate_path<ObstrLyt> get_path_and_obstruct(ObstrLyt& lyt, const tile<ObstrLyt>& start_tile,
+                                                           const tile<ObstrLyt>& end_tile)
+    {
+        using dist = twoddwave_distance_functor<ObstrLyt, uint64_t>;
+        using cost = unit_cost_functor<ObstrLyt, uint8_t>;
+        static a_star_params params{};
+        params.crossings = !ps.planar_optimization;
```

Comment:
Even though `params` is `static` here, this assignment is executed every time the function is called. Maybe you want to move it inside `params`' constructor.

Reply:
And also maybe you want to rename `params` to state that they are related to A*.

Reply:
Ah, of course. Because `ps.planar_optimization` is only known at call time. However, `a_star_params` is already instantiated when the program starts.

Btw, avoid naming collision. The variable has the same name as its type right now.